src/securityPolicy.c

```diff
@@ -242,11 +245,18 @@ security_policy_t policyForSignTxInit(
 
 		// mint must not be combined with pool registration certificates
 		DENY_IF(includeMint);
+		DENY_IF(numCollaterals != 0 || numRequiredSigners != 0);
 		break;
 
 	case SIGN_TX_SIGNINGMODE_ORDINARY_TX:
 	case SIGN_TX_SIGNINGMODE_MULTISIG_TX:
-		// no additional validation
+		DENY_IF(numCollaterals != 0 || numRequiredSigners != 0);
+		break;
+
+	case SIGN_TX_SIGNINGMODE_PLUTUS_TX:
+		WARN_IF(numCollaterals != 0);
+		WARN_IF(numCollaterals == 0);
```

Comment:
If I understand the UI code in `signTx.c` correctly, `WARN_IF(numCollaterals != 0)` is here only because we want to show the `Plutus script will be run` warning when both collaterals and script data hash are present in `PLUTUS_TX` signing mode. Wouldn't something like this be more semantically correct then?
```
case SIGN_TX_SIGNINGMODE_PLUTUS_TX:
	// Plutus script will not be executed without collaterals
	WARN_IF(numCollaterals == 0);

	// Plutus script will not be executed without script data hash
	WARN_UNLESS(includeScriptDataHash);

	// warn the user about Plutus script execution
	WARN();
	break;
```

Reply:
Yeah, the collaterals > 0 warning is functionally the same as the signingmode_plutus in normal cases, so you are right.